File: test_technique/src/app/models/product.ts
export class Product {
    name?: string;
    code?: string;
    category?: string;
    inventory_status?: string;
    price?: number;
    quantity?: number;
    rating?: number;
    description?: string;
    image?: string;
    internal_reference?: string;
    shell_id?: number;
  
    constructor(init?: Partial<Product>) {
      Object.assign(this, init);
    }
  }
  

File: back/Procuct_Back-end/src/main/java/com/backend/procuct_backend/dto/Product.java
package com.backend.procuct_backend.dto;

import com.backend.procuct_backend.entitie.ProductEntity;
import lombok.Data;

import java.time.Instant;

@Data
public class Product {
    private int id;
    private String code;
    private String name;
    private String description;
    private String image;
    private String category;
    private Integer price;
    private int quantity;
    private String internalReference;
    private Long shellId;
    private ProductEntity.InventoryStatus inventoryStatus;
    private int rating;
    private Instant createdAt;
    private Instant updatedAt;
    private Integer userId;
}